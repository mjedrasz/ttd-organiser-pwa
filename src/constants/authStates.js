const SIGNED_IN = 'signedIn';
const SIGN_IN = 'signIn';
const CONFIRM_SIGN_UP = 'confirmSignUp';
const SIGN_UP = 'signUp';
const FORGOT_PASSWORD = 'forgotPassword';
const RESET_PASSWORD = 'resetPassword';
const VERIFY_CONTACT = 'verifyContact';

export {
    SIGNED_IN,
    SIGN_IN,
    CONFIRM_SIGN_UP, 
    SIGN_UP,
    FORGOT_PASSWORD,
    RESET_PASSWORD,
    VERIFY_CONTACT
};