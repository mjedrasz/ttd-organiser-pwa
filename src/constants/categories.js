import { I18n } from "aws-amplify";

const MUSIC = "MUSIC";
const DANCE = "DANCE";
const ARTS = "ARTS";
const CRAFTS = "CRAFTS";
const SPORT = "SPORT";
const CULTURE = "CULTURE";
const OUTDOOR = "OUTDOOR";
const CINEMA = "CINEMA";
const THEATER = "THEATER";
const PERSONAL_DEVELOPMENT = "PERSONAL_DEVELOPMENT";
const PARTY = "PARTY";
const OTHER = "OTHER";

const labels = {
    [MUSIC]: I18n.get("Music"),
    [DANCE]: I18n.get("Dance"),
    [ARTS]: I18n.get("Arts"),
    [CRAFTS]: I18n.get("Crafts"),
    [SPORT]: I18n.get("Sport"),
    [OUTDOOR]: I18n.get("Outdoor"),
    [CULTURE]: I18n.get("Culture"),
    [CINEMA]: I18n.get("Cinema"),
    [THEATER]: I18n.get("Theater"),
    [PERSONAL_DEVELOPMENT]: I18n.get("Personal development"),
    [PARTY]: I18n.get("Party"),
    [OTHER]: I18n.get("Other")
};
const keys = [
    MUSIC,
    DANCE,
    ARTS,
    CRAFTS,
    SPORT,
    OUTDOOR,
    CULTURE,
    CINEMA,
    THEATER,
    PERSONAL_DEVELOPMENT,
    PARTY,
    OTHER];
    
export {
    keys,
    labels,
    MUSIC,
    DANCE,
    ARTS,
    CRAFTS,
    SPORT,
    OUTDOOR,
    CULTURE,
    CINEMA,
    THEATER,
    PERSONAL_DEVELOPMENT,
    PARTY,
    OTHER
};