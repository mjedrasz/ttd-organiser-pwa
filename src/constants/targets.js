import { I18n } from "aws-amplify";

const ADULT = "ADULT";
const FAMILY = "FAMILY";
const CHILD = "CHILD";

const labels = {
    [ADULT]: I18n.get("Adult"),
    [FAMILY]: I18n.get("Family"),
    [CHILD]: I18n.get("Child")
};

const keys = [
    ADULT,
    FAMILY,
    CHILD
];
    
export {
    keys,
    labels,
    ADULT,
    FAMILY,
    CHILD
};