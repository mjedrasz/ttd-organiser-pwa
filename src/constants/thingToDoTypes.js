import { I18n } from "aws-amplify";

const EVENT = "EVENT";
const RECURRING_EVENT = "RECURRING_EVENT";
const PLACE = "PLACE";

const labels = {
    [EVENT]: I18n.get("Event"),
    [RECURRING_EVENT]: I18n.get("Recurring event"),
    [PLACE]: I18n.get("Place")
};

const keys = [
    EVENT,
    RECURRING_EVENT,
    PLACE
];
    
export {
    keys,
    labels,
    EVENT,
    RECURRING_EVENT,
    PLACE
};