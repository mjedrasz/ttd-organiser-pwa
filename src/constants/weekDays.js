import { I18n } from "aws-amplify";

const MON = "MON";
const TUE = "TUE";
const WED = "WED";
const THU = "THU";
const FRI = "FRI";
const SAT = "SAT";
const SUN = "SUN";

const labels = {
    [MON]: I18n.get("Mon"),
    [TUE]: I18n.get("Tue"),
    [WED]: I18n.get("Wed"),
    [THU]: I18n.get("Thu"),
    [FRI]: I18n.get("Fri"),
    [SAT]: I18n.get("Sat"),
    [SUN]: I18n.get("Sun")
};

const keys = [
    MON,
    TUE,
    WED,
    THU,
    FRI,
    SAT,
    SUN
];
    
export {
    keys,
    labels,
    MON,
    TUE,
    WED,
    THU,
    FRI,
    SAT,
    SUN
};