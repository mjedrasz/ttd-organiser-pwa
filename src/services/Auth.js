import Auth from '@aws-amplify/auth';
import JS from '@aws-amplify/core/lib/JS';

import { SIGNED_IN, VERIFY_CONTACT } from '../constants/authStates';

const GOOGLE_PROVIDER = 'Google';
const FACEBOOK_PROVIDER = 'Facebook';

export class AuthClass {

    confirmSignUp = async (username, code) => {
        await Auth.confirmSignUp(username, code);
    };

    resendCodeSignUp = async (username) => {
        await Auth.resendSignUp(username);
    };

    forgotPassword = async (username) => {
        await Auth.forgotPassword(username)
    };

    resetPassword = async (username, code, password) => {
        await Auth.forgotPasswordSubmit(username, code, password);
    };

    resendCode = async (username) => {
        await Auth.forgotPassword(username);
    };

    signIn = async (email, password) => {
        const user = await Auth.signIn(email, password);
        return await this.checkContact(user);
    };

    checkContact = async (user) => {
        const data = await Auth.verifiedContact(user);
        if (!JS.isEmpty(data.verified)) {
            return {
                state: SIGNED_IN,
                data: user
            };
        } else {
            return {
                state: VERIFY_CONTACT,
                data: { ...user, ...data }
            };
        }
    };

    socialSignIn = (provider) => {
        const config = Auth.configure();

        const {
            userPoolWebClientId,
            oauth: {
                scope,
                domain,
                redirectSignIn,
                responseType
            }
        } = config;
        const url = `https://${domain}/oauth2/authorize?identity_provider=${provider}&redirect_uri=${redirectSignIn}&response_type=${responseType}&client_id=${userPoolWebClientId}&scope=${scope.join('+')}`;
        window.location.assign(url);
    }

    googleSignIn = () => {
        this.socialSignIn(GOOGLE_PROVIDER);
    }

    facebookSignIn = () => {
        this.socialSignIn(FACEBOOK_PROVIDER);
    }

    signUp = async (username, password, attributes = {}) =>
        await Auth.signUp({
            username,
            password,
            attributes
        });


    signOut = async () => await Auth.signOut({ global: true });
}

const AuthService = new AuthClass();

export default AuthService;