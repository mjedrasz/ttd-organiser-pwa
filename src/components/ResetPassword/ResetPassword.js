import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import CircularProgress from '@material-ui/core/CircularProgress';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import { Logger } from '@aws-amplify/core';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import I18n from "@aws-amplify/core/lib/I18n";
import useStyles from './styles';
import Toast from '../Toast/Toast';
import { useForm } from '../../hooks/useForm';
import { useToast } from '../../hooks/useToast';
import { useSubmit } from '../../hooks/useSubmit';
import Auth from '../../services/Auth';
import * as AuthState from '../../constants/authStates';

const logger = new Logger('ResetPassword');

export default (props) => {

  const classes = useStyles();

  const { useInput, isValid, values, preventValidation } = useForm();

  const [toast, showToast] = useToast();

  const resendCode = async () => {
    try {
      await Auth.resendCode(props.authData);
      showToast(I18n.get('Operation succeeded'), 'success');
    } catch (err) {
      logger.error("Resend code failed", err);
      showToast(I18n.get('Resend code failed'), 'error');
    }
  };

  const resetPassword = async () => {
    try {
      const { code, password } = values;
      await Auth.resetPassword(props.authData, code, password);
      props.onStateChange(AuthState.SIGN_IN, props.authData);
    } catch (err) {
      logger.error("Reset password failed", err);
      showToast(I18n.get('Reset password failed'), 'error');
    }
  };

  const [handleResetPassword, submittingResetPassword] = useSubmit(resetPassword);
  const [handleResendCode, submittingResendCode] = useSubmit(resendCode);

  const codeInput = useInput('code', '', {
    isRequired: I18n.get('The field is required')
  });

  const passwordInput = useInput('password', '', {
    isRequired: I18n.get('The field is required'),
    isMinLength: { length: 8, message: I18n.get('The passwords must be at least 8 characters long') }
  });

  return [AuthState.RESET_PASSWORD].includes(props.authState) && (
    <>
      <Toast {...toast}/>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            {I18n.get("Reset password")}
          </Typography>
          <form className={classes.form} noValidate>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="confirmation-code"
              type="text"
              label={I18n.get("Confirmation Code")}
              autoComplete="off"
              id="confirmation-code"
              {...codeInput}
            />
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="new-password"
              label={I18n.get("New password")}
              type="password"
              id="new-password"
              autoComplete="off"
              {...passwordInput}
            />
            <div className={classes.wrapper}>
              <Button
                onClick={handleResetPassword}
                fullWidth
                disabled={!isValid || submittingResetPassword}
                variant="contained"
                color="primary"
                className={classes.submit}
              >
                {I18n.get("Verify")}
              </Button>
              {submittingResetPassword && <CircularProgress size={24} className={classes.submitProgress} />}
            </div>
            <Grid container>
              <Grid item xs>
                <Link href="#" variant="body2" onClick={handleResendCode} onMouseDown={preventValidation}>
                  {I18n.get("Lost your code? Resend Code")}
                  {submittingResendCode && <CircularProgress size={16} />}
                </Link>
              </Grid>
            </Grid>
          </form>
        </div>
        
      </Container>
    </>
  );
}