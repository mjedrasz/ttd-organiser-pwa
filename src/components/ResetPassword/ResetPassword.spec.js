import React from 'react';
import renderer from 'react-test-renderer';
import ResetPassword from './ResetPassword';
import { render, fireEvent, act, waitForElement } from '@testing-library/react';
import Auth from '../../services/Auth';
import I18n from "@aws-amplify/core/lib/I18n";
import * as AuthState from '../../constants/authStates';

jest.mock('../../services/Auth', () => ({
  resetPassword: jest.fn(),
  resendCode: jest.fn()
}));

// jest.mock('@material-ui/core/TextField', () =>
//   jest.fn(({ value, label, helperText, onChange, onBlur, onFocus }) => (
//     <div>
//     <label>
//       {label}
//       <input onChange={onChange} onBlur={onBlur} onFocus={onFocus} value={value} />
//     </label>
//     <div>{helperText}</div>
//     </div>))
// );

describe('ResetPassword', () => {
  test('snapshot renders correctly', () => {

    const component = renderer.create(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('not rendered when authState is not signUp', () => {

    const { queryAllByText } = render(<ResetPassword authState="other state" />);

    expect(queryAllByText(I18n.get('Reset password')).length).toEqual(0);
  });

  it('rendered when authState is signUp', () => {

    const { queryAllByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    expect(queryAllByText(I18n.get('Reset password')).length).toBeGreaterThan(0);
  });

  it('submit button disabled when fields empty', () => {

    const { getByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);
    const submit = getByText(I18n.get('Verify'));

    expect(submit).toBeDisabled();
  });

  it('submit button enabled when fields with valid values', () => {

    const { getByLabelText, getByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const password = getByLabelText(`${I18n.get('New password')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(code, { target: { value: '123456' } });
    fireEvent.change(password, { target: { value: 'secret123' } });

    expect(submit).not.toBeDisabled();
  });

  it('submit button disabled when code is empty', () => {

    const { getByLabelText, getByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(code, { target: { value: '' } });

    expect(submit).toBeDisabled();
  });

  it('submit button disabled when password is empty', () => {

    const { getByLabelText, getByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const password = getByLabelText(`${I18n.get('New password')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(password, { target: { value: '' } });

    expect(submit).toBeDisabled();
  });

  it('display error messages when code is empty and on blur', () => {

    const { getByLabelText, queryByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);

    expect(queryByText(I18n.get('The field is required'))).not.toBeInTheDocument();

    fireEvent.focus(code, {});
    fireEvent.change(code, { target: { value: '' } });
    fireEvent.blur(code, {});

    expect(queryByText(I18n.get('The field is required'))).toBeInTheDocument();
  });

  it('display error messages when password is empty and on blur', () => {

    const { getByLabelText, queryByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const password = getByLabelText(`${I18n.get('New password')} *`);

    expect(queryByText(I18n.get('The field is required'))).not.toBeInTheDocument();

    fireEvent.focus(password, {});
    fireEvent.change(password, { target: { value: '' } });
    fireEvent.blur(password, {});

    expect(queryByText(I18n.get('The field is required'))).toBeInTheDocument();
  });

  it('do not display error messages when typing', () => {

    const { getByLabelText, queryByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const password = getByLabelText(`${I18n.get('New password')} *`);

    expect(queryByText(I18n.get('The field is required'))).not.toBeInTheDocument();

    fireEvent.focus(code, {});
    fireEvent.change(code, { target: { value: '' } });
    fireEvent.focus(password, {});
    fireEvent.change(password, { target: { value: '' } });

    expect(queryByText(I18n.get('The field is required'))).not.toBeInTheDocument();
  });

  it('error message when confirm sign up failed', async () => {

    Auth.resetPassword.mockImplementation(
      () => {
        throw Error('Unable to reset password');
      });

    const { queryByText, getByLabelText, getByText} = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const password = getByLabelText(`${I18n.get('New password')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(code, { target: { value: '123456' } });
    fireEvent.change(password, { target: { value: 'secret123' } });

    expect(queryByText(I18n.get('Reset password failed')), 'no error message yet').not.toBeInTheDocument();

    await act(async () => {
      submit.click();
    });

    expect(queryByText(I18n.get('Reset password failed'))).toBeInTheDocument();
  });

  it('error message when resending code failed', async () => {

    Auth.resendCode.mockImplementation(
      () => {
        throw Error('Unable to resend code');
      });

    const { getByText, queryByText } = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" />);

    const resendCodeLink = getByText(`${I18n.get('Lost your code? Resend Code')}`);

    expect(queryByText(I18n.get('Resend code failed')), 'no error message yet').not.toBeInTheDocument();
  
    await act(async () => {
      resendCodeLink.click();
    });

    expect(queryByText(I18n.get('Resend code failed'))).toBeInTheDocument();
  });

  it('should change state to "signIn" on submit', async () => {

    const codeValue = '123456';
    const emailValue = 'email@example.com';
    const passwordValue = 'secret123';

    const mockOnStateChange = jest.fn((state, _) => {
      expect(state).toEqual('signIn');
    });

    Auth.resetPassword.mockImplementation(
      (username, code, password) => {
        expect(username).toEqual(emailValue);
        expect(password).toEqual(passwordValue);
        expect(code).toEqual(codeValue);
      });

    const { getByLabelText, getByText } = render(<ResetPassword
      authState={AuthState.RESET_PASSWORD}
      authData={emailValue} 
      onStateChange={mockOnStateChange} />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const password = getByLabelText(`${I18n.get('New password')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(code, { target: { value: codeValue } });
    fireEvent.change(password, { target: { value: passwordValue } });

    await act(async () => {
      submit.click();
    });

    expect(mockOnStateChange.mock.calls.length).toBe(1);

  });

  it('should display progress and disable submit button while submitting', async () => {

    const mockOnStateChange = jest.fn((state, _) => { });

    Auth.resetPassword.mockImplementation(
      async () => {});

    const { getByLabelText, getByRole, queryByRole, getByText} = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" onStateChange={mockOnStateChange} />);

    const code = getByLabelText(`${I18n.get('Confirmation Code')} *`);
    const password = getByLabelText(`${I18n.get('New password')} *`);
    const submit = getByText(I18n.get('Verify'));

    fireEvent.change(code, { target: { value: '123456' } });
    fireEvent.change(password, { target: { value: 'secret123' } });

    expect(submit).not.toBeDisabled();
    expect(queryByRole(`progressbar`)).not.toBeInTheDocument();

    fireEvent.click(submit);

    await act(async () => {
      expect(submit).toBeDisabled();
      await waitForElement(() => getByRole(`progressbar`));
    });

    expect(submit).not.toBeDisabled();
    expect(queryByRole(`progressbar`)).not.toBeInTheDocument();

  });

  it('should display progress when resending code', async () => {

    const mockOnStateChange = jest.fn((state, _) => { });

    Auth.resendCode.mockImplementation(
      async () => {});

    const { getByRole, queryByRole, getByText} = render(<ResetPassword authState={AuthState.RESET_PASSWORD} authData="email@example.com" onStateChange={mockOnStateChange} />);

    const resendCodeLink = getByText(`${I18n.get('Lost your code? Resend Code')}`);

    expect(queryByRole(`progressbar`)).not.toBeInTheDocument();

    fireEvent.click(resendCodeLink);

    await act(async () => {
      await waitForElement(() => getByRole(`progressbar`));
    });

    expect(queryByRole(`progressbar`)).not.toBeInTheDocument();

  });

});