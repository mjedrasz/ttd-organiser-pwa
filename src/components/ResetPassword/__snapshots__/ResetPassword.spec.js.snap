// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword snapshot renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="none"
        >
          <path
            d="M0 0h24v24H0V0z"
          />
          <path
            d="M0 0h24v24H0V0z"
            opacity=".87"
          />
        </g>
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Reset password
    </h1>
    <form
      className="makeStyles-form-3"
      noValidate={true}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="confirmation-code"
          id="confirmation-code-label"
        >
          Confirmation Code
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="confirmation-code"
            name="code"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-128 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-129"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="new-password"
          id="new-password-label"
        >
          New password
          <span
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="new-password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-128 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-129"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        className="makeStyles-wrapper-5"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Verify
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
          >
            Lost your code? Resend Code
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
